cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1519,14 +1543,14 @@ void reader::impl::preprocess_subpass_pages(read_mode mode, size_t chunk_read_li
       max_col_row--;
     }
 
-    max_row = min(max_row, max_col_row);
+    max_row = std::min<size_t>(max_row, max_col_row);
 
     first_page_index += subpass.column_page_count[idx];
   }
   subpass.skip_rows   = pass.skip_rows + pass.processed_rows;
   auto const pass_end = pass.skip_rows + pass.num_rows;
-  max_row             = min(max_row, pass_end);
-  CUDF_EXPECTS(max_row >= subpass.skip_rows, "Unexpected short subpass", std::underflow_error);
+  max_row             = std::min<size_t>(max_row, pass_end);
+  CUDF_EXPECTS(max_row > subpass.skip_rows, "Unexpected short subpass", std::underflow_error);
```

Comment:
Is this right?  If skip_rows == 2, the first possible row is index 2.

Reply:
We don't want to end up with subpass.num_rows = max_row - subpass.skip_row = 0 or underflow (a huge size_t number) as we may either go in an infinite loop (with zero) or an exception (underflowed size_t)

Reply:
I guess I was reading `max_row` as "maximum row index" (inclusive), but it looks like it's more like "end".